src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -1627,5 +1627,338 @@ Class C
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public readonly ref struct [|S|]
+{{
+}}");
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic, $@"#Region ""{FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null""
+' {CodeAnalysisResources.InMemoryAssembly}
+#End Region
+
+Imports System
+Imports System.Runtime.CompilerServices
+
+<IsByRefLikeAttribute> <IsReadOnlyAttribute> <Obsolete(""Types with embedded references are not supported in this version of your compiler."", True)>
+Public Structure [|S|]
+End Structure");
+        }
+
+
+        [WorkItem(34650, "https://github.com/dotnet/roslyn/issues/34650")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestReadOnlyMethod()
+        {
+            var metadataSource = @"
+public struct S
+{
+    public readonly void M() {}
+}
+";
+            var symbolName = "S.M";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, languageVersion: "Preview",
+                expected: $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public struct S
+{{
+    public readonly void [|M|]();
+}}");
+
+            // TODO: how to use preview features in metadata source but display in VB?
+            //            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic,
+            //                expected: $@"#Region ""{FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null""
+            //' {CodeAnalysisResources.InMemoryAssembly}
+            //#End Region
+
+            //Imports System
+            //Imports System.Runtime.CompilerServices
+
+            //Public Structure S
+            //    <IsReadOnlyAttribute>
+            //    Sub [|M|]();
+            //End Structure");
+        }
+
+        [WorkItem(34650, "https://github.com/dotnet/roslyn/issues/34650")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestStructProperty_ReadOnly()
+        {
+            var metadataSource = @"
+public struct S
+{
+    public int P { get; }
+}
+";
+            var symbolName = "S.P";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, languageVersion: "Preview",
+                expected: $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public struct S
+{{
+    public readonly int [|P|] {{ get; }}
```

Comment:
It's intentional we'll show what the compiler added here?

Reply:
Yeah. A lot of manually-implemented getters will be non-readonly and it's not possible to tell from MetadataAsSource whether a property was auto- or manually-implemented.